src/App.jsx

```diff
@@ -14,45 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const visibleGoods = [...goodsFromServer];
+
+  if (isReverse && !sortField) {
+    visibleGoods.reverse();
+  }
+
+  if (sortField) {
+    visibleGoods.sort((a, b) => {
+      let result;
+
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          result = a.localeCompare(b);
+          break;
+
+        case SORT_FIELD_LENGTH:
+          result = a.length - b.length;
+
+          if (result === 0) {
+            result = a.localeCompare(b);
+          }
+
+          break;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        default:
+          return 0;
+      }
+
+      if (isReverse) {
+        return -result;
+      }
+
+      return result;
+    });
+  }
```

Comment:
1. This slightly overcomplicated logic. This should be enough
```
visibleGoods.sort((a, b) => {
    switch (sortField) {
      case SORT_FIELD_ALPHABET:
        return a.localeCompare(b);
      case SORT_FIELD_LENGTH:
        return a.length - b.length;
      default:
        return 0;
    }
})
```
2. Let's move sorting to separate function outside of the component

Reply:
I add 

```
if (result === 0) {
  result = a.localeCompare(b);
}
```
cause without it, I got

Dumplings
Ice cream // must be the first in the task
Garlic
Carrot
Honey
Bread
Apple
Fish
Eggs
Jam